test/crawler_web/controllers/api/v1/user_registration_controller_test.exs

```diff
@@ -0,0 +1,131 @@
+defmodule CrawlerWeb.Api.V1.UserRegistrationControllerTest do
+  use CrawlerWeb.ConnCase, async: true
+
+  alias Crawler.Account.Guardian
+
+  describe "POST create/2" do
+    test "given valid user credentials, returns access token", %{conn: conn} do
+      email = "new_email@gmail.com"
+
+      conn =
+        post(conn, Routes.api_v1_user_registration_path(conn, :create), %{
+          email: email,
+          password: "12345678"
+        })
+
+      assert %{
+               "data" => %{
+                 "attributes" => %{
+                   "email" => ^email,
+                   "token" => _
+                 },
+                 "id" => _,
+                 "relationships" => %{},
+                 "type" => "user_registration"
+               },
+               "included" => []
+             } = json_response(conn, 200)
+    end
+
+    test "given invalid user credentials, returns error response", %{conn: conn} do
+      conn = post(conn, Routes.api_v1_user_registration_path(conn, :create), %{})
+
+      assert json_response(conn, 422) == %{
+               "errors" => [
+                 %{
+                   "code" => "unprocessable_entity",
+                   "detail" => gettext("Email or password is missing")
+                 }
+               ]
+             }
+    end
+
+    test "given existing user email, returns error response", %{conn: conn} do
+      %{email: email, password: password} = insert(:user)
+
+      conn =
+        post(conn, Routes.api_v1_user_registration_path(conn, :create), %{
+          email: email,
+          password: password
+        })
+
+      assert json_response(conn, 422) == %{
```

Comment:
This is not 422 status for field validations (and below as well)

```suggestion
      assert json_response(conn, 400) == %{
```

Reply:
Normally we return 422 for this case :D

Reply:
Oh, just updated it to 400 (bad_request) :( Should I revert them back?

Reply:
It's ok @vnntsu not a big deal here :)

Reply:
> Normally we return 422 for this case :D

I don't see any convention for that in the compass 🤔 

My suggestion is based on this RFC: https://www.rfc-editor.org/rfc/rfc7807

Which suggests to use 400 response for validation errors. 

Also here 422 status even is not mentioned.

422 is from webdav extension: http://greenbytes.de/tech/webdav/rfc4918.html#rfc.section.11.2

Reply:
422 just come from the Rails convention FYI @Nihisil , whenever we have the error on the model validation/saving/creating, the Rails returns 422 status which Unprocessable Entity and the team keep that convention in Elixir projects too

Reply:
@andyduong1920 thank you for clarifying, I wasn't aware that we have this convention 😓

Reply:
@Nihisil it's not an official convention but we used that in several client projects :D